### ISSUE TITLE
Official Docs Contribution

### ISSUE THREAD

**Issue by heywhy** [heywhy/ex_elasticlunr#10]
I will be happy to have anyone contribute to the official docs for the project. Having comprehensive docs at [hex.pm](https://hexdocs.pm) is desired, so modules need to be updated.

**Comment by TwistingTwists** [comment]
Hey. 
This is a great project! 
I have used - [Meilisearch](https://www.meilisearch.com/) previously - a similar project in Rust.

Bringing a similar project to Elixir, brings me delight.

I am willing to contribute to the docs. Can you point me where to start?

**Comment by heywhy** [comment]
@TwistingTwists I'm glad to have you contribute to the docs or any other part of the project. To get started, I will suggest you start by updating the documentation (including examples where applicable) for each module based on it behavior or logic contained in it. Do not hesitate to reach out (via email) in case you are unsure of what a module does or how it contributes to the project. Note that you will need to familiarize yourself with the internal workings of the library so as to help you write more informative documentation.

I hope my suggestion isn't vague, do let me know if you have a better approach that will get us going with the documentation.

Thank you for wanting to contribute to the project, I really appreciate it.

**Comment by drannex42** [comment]
+1 on Docs, but also the code should be commented! I read/write in Elixir but some of it was a bit hard to follow (not too hard, but comments are always great!). 

Using the [@doc ](https://hexdocs.pm/elixir/1.12/writing-documentation.html) macro would aid in inline understanding (and future proofing) the code, and help with adding some quick and simple documentation.

Thanks for your work on this @heywhy will def be using this soon!

**Comment by heywhy** [comment]
@drannex42 thank you for the feedback. your suggestion is what we’re working on already, module docs will be coming in before any other external docs.

Actually, the goal is to have an informational documentation on hexdocs.

**Comment by drannex42** [comment]
Great to hear! Happy that it's already underway. Again, really great project you have here.

**Comment by heywhy** [comment]
No worries. I posted the issue myself to solicit for contributors to help.😅
